addons/l10n_sa_edi/models/account_move.py

```diff
@@ -264,6 +271,26 @@ def _get_l10n_sa_totals(self):
             'total_tax': invoice_vals['vals']['tax_total_vals'][-1]['tax_amount'],
         }
 
+    def _retry_edi_documents_error(self):
+        """
+            Hook to reset the chain head error prior to retrying the submission
+        """
+        self.filtered(lambda m: m.country_code == 'SA').write({'l10n_sa_edi_chain_head_id': False})
+        return super()._retry_edi_documents_error()
+
+    def action_show_chain_head(self):
+        """
+            Action to show the chain head of the invoice
+        """
+        self.ensure_one()
+        return {
+            'name': _("Chain Head"),
+            'type': 'ir.actions.act_window',
+            'res_model': 'account.move',
+            'view_mode': 'form',
+            'res_id': self.l10n_sa_edi_chain_head_id.id,
+        }
```

Comment:
i think you can use _get_records_action() instead :smile: 
```suggestion
        return self.l10n_sa_edi_chain_head_id._get_records_action(name=_("Chain Head"))
```